travis-ci/vmtest/checkout_latest_kernel.sh

```diff
@@ -8,12 +8,22 @@ CWD=$(pwd)
 LIBBPF_PATH=$(pwd)
 REPO_PATH=$1
 
-BPF_NEXT_ORIGIN=https://git.kernel.org/pub/scm/linux/kernel/git/bpf/bpf-next.git
-LINUX_SHA=$(cat ${LIBBPF_PATH}/CHECKPOINT-COMMIT)
-SNAPSHOT_URL=https://git.kernel.org/pub/scm/linux/kernel/git/bpf/bpf-next.git/snapshot/bpf-next-${LINUX_SHA}.tar.gz
+KERNEL_ORIGIN=${KERNEL_ORIGIN:-https://git.kernel.org/pub/scm/linux/kernel/git/bpf/bpf-next.git}
+KERNEL_BRANCH=${KERNEL_BRANCH:-CHECKPOINT}
+if [[ "${KERNEL_BRANCH}" = 'CHECKPOINT' ]]; then
+  echo "using CHECKPOINT sha1"
+  LINUX_SHA=$(cat ${LIBBPF_PATH}/CHECKPOINT-COMMIT)
+else
+  echo "using ${KERNEL_BRANCH} sha1"
+  LINUX_SHA=$(git ls-remote ${KERNEL_ORIGIN} ${KERNEL_BRANCH} | awk '{print $1}')
+fi
+SNAPSHOT_URL=${KERNEL_ORIGIN}/snapshot/bpf-next-${LINUX_SHA}.tar.gz
```

Comment:
this probably won't work if someone will try to fetch kernel from non-bpf-next tree, but we have a fallback logic below, so it's a good start